apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,332 @@
+			}
+			read_len += valread;
+		}
+		strncpy(buffer + read_len, ptr_msg, sizeof(SERVER_MSG));
+		total_send = strlen(buffer);
+		while (total_send) {
+			valsend = send(client_socket, buffer + send_len, strlen(buffer) - send_len, 0);
+			printf("server send: %s\n", buffer);
+			if (valsend == -1) {
+				close(client_socket);
+				close(server_socket);
+				return (void *)pret;
+			}
+			total_send -= valsend;
+			if (total_send < 0) {
+				close(client_socket);
+				close(server_socket);
+				pret = ERROR;
+				return (void *)pret;
+			}
+			send_len += valsend;
+		}
+		sleep(SERVER_WAIT_TIME);
+		ret = close(client_socket);
+		if (ret != OK) {
+			close(server_socket);
+			pret = ERROR;
+			return (void *)pret;
+		}
+		printf("Server thread : finished with client\n");
+	}
+	ret = close(server_socket);
+	if (ret != OK) {
+		return (void *)pret;
+	}
+	printf("Server completed\n");
+	return (void *)OK;
+}
+
+/**
+* @fn               :client_connect
+* @description      :starts client
+* @return           :void
+*/
+static void *client_connect(void *ptr_id)
+{
+	int *id;
+	id = (int *)ptr_id;
+	int sock = 0;
+	int ret;
+	int valread = 0;
+	int valsend = 0;
+	struct sockaddr_in serv_addr;
+	char client_msg[BUFF_LEN];
+	char buffer[BUFF_LEN] = {0};
+	int total_read;
+	int total_send;
+	int pret = OK;
+	int read_len = 0;
+	int send_len = 0;
+
+	printf("Client thread started...\n");
+	sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock < 0) {
+		return (void *)pret;
+	}
+	memset(&serv_addr, '0', sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	serv_addr.sin_port = htons(g_port);
+	ret = connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+	if (ret != OK) {
+		close(sock);
+		return (void *)ret;
+	}
+	strncpy(client_msg, CLIENT_MSG, sizeof(CLIENT_MSG));
+	client_msg[strlen(client_msg) - 2] = '0' + (*id);
+	total_send = strlen(client_msg);
+	while (total_send) {
+		valsend = send(sock, client_msg + send_len, strlen(client_msg) - send_len, 0);
+		printf("client send: %s\n", client_msg);
+		if (valsend == -1) {
+			close(sock);
+			return (void *)pret;
+		}
+		total_send -= valsend;
+		if (total_send < 0) {
+			close(sock);
+			pret = ERROR;
+			return (void *)pret;
+		}
+		send_len += valsend;
+	}
+	total_read = strlen(client_msg) + strlen(SERVER_MSG);
+	memset(buffer, 0, BUFF_LEN);
+	while (total_read) {
+		valread = read(sock, buffer + read_len, BUFF_LEN - read_len);
+		printf("client read: %s\n", buffer);
+		if (valread <= 0) {
+			close(sock);
+			pret = ERROR;
+			return (void *)pret;
+		}
+		total_read -= valread;
+		if (total_read < 0) {
+			close(sock);
+			pret = ERROR;
+			return (void *)pret;
+		}
+		read_len += valread;
+	}
+	if (strncmp(buffer, client_msg, strlen(client_msg)) != 0) {
+		close(sock);
+		pret = ERROR;
+		return (void *)pret;
+	}
+	if (strncmp(buffer + strlen(client_msg), SERVER_MSG, strlen(SERVER_MSG)) != 0) {
+		close(sock);
+		pret = ERROR;
```

Comment:
pret is a local variable, so it would be invalidated when function is finished. so it can points invalid memory space.
so I think it is more safe to use malloc

Reply:
changes  are done as per your suggestion.  please review